streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBRangeIterator.java

```diff
@@ -63,17 +65,20 @@
         final KeyValue<Bytes, byte[]> next = super.makeNext();
         if (next == null) {
             return allDone();
+        } else if (rawLastKey == null) {
+            return next;
+
         } else {
             if (forward) {
-                if (comparator.compare(next.key.get(), rawLastKey) < 0) {
+                if (rawLastKey != null && comparator.compare(next.key.get(), rawLastKey) < 0) {
```

Comment:
how can `rawLastKey` be null here (c.f. branch above)?

Reply:
Good catch. Yes this check can be eliminated now that there is a common null check in the beginnin.